core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -245,6 +244,14 @@ protected void doStop() throws Exception {
       context.setBaseResource(staticResources);
     }
 
+    if(config.getBoolean(RestConfig.ENABLE_GZIP_COMPRESSION_CONFIG)) {
+      FilterHolder gzipFilter = new FilterHolder(AsyncGzipFilter.class);
+      // do not check if .gz file already exists for the requested resource
+      gzipFilter.setInitParameter("checkGzExists", "false");
```

Comment:
Is this never useful or just not important for the use case this is being added for? Presumably for static assets this can be useful to avoid re-compressing?